bgpd/bgp_network.c

```diff
@@ -459,7 +465,8 @@ static int bgp_bind(struct peer *peer)
 				      : (peer->ifname ? peer->ifname
 						      : peer->bgp->name));
 	else
-		name = peer->ifname ? peer->ifname : peer->bgp->name;
+		name = peer->ifname ? peer->ifname :
+			(peer->bgp != NULL ? peer->bgp->name : NULL);
```

Comment:
using scan-build make, I encountered this issue.
do you see any solution to avoid that scan-build ignores that issue ?

Reply:
If this fixes an existing warning, then the fix should be provided in a separate commit. If the warning was introduced by your changes, then it's worth investigating the cause of it instead of just silencing the warning IMO. I'll take a closer look at this tomorrow.

Reply:
ok. done.

Reply:
I couldn't reproduce this scan-build warning using clang 3.8.1. Which version of clang are you using? I'd like to investigate this warning in more detail.

Reply:
I use clang 3.8. you are right I remove the change.